pygments/lexers/modeling.py

```diff
@@ -363,3 +364,66 @@ def analyse_text(text):
             return 1.0
         else:
             return 0.0
+            
+class UmpleLexer(JavaLexer):
+  """
+    Pygments lexer for Umple <https://cruise.umple.org/umple/> source code
+
+    .. versionadded::1.31 
+  """
+
+
+  name="Umple"
+  aliases=['umple']
+  filenames = ['*.ump', '*.umpt']
+  
+  tokens = {
+    'root': [
+      (r'(active|after|all|around|atomic|attr|attribute|before|'
+      r'custom|deactivate|during|emit|entry|execute|exit|generate|generated|hops|'
+      r'include|includeFilter|inner|isA|key|model|[oO]n|[oO]ff|pooled|position|post|prefix|queued|regex|singleton|'
+      r'sorted|sub|subclass|suffix|super|superclass|top|trace|unspecified)\b',
```

Comment:
This (and the other word lists) should use the `words` function which optimizes the regex.

Reply:
My student has made this change, and the others mentioned in your review, so hopefully you can approve this